scripts/build_llvm_project.py

```diff
@@ -0,0 +1,139 @@
+                        help=f'specify build directory (default: {settings.LLVM_DIR}/%(default)s)')
+  buildopt.add_argument('--clean', action='store_true', default=False,
+                        help='clean build')
+  buildopt.add_argument('--build-type', metavar='TYPE', default='Release',
+                        help='set build type (default: %(default)s)')
+  buildopt.add_argument('--debug', dest='build_type', action='store_const', const='Debug', default=argparse.SUPPRESS,
+                        help='debug build (alias for --build-type %(const)s)')
+  buildopt.add_argument('--cmake-param', metavar='OPT', action='append', default=[],
+                        help='add custom argument to CMake')
+  buildopt.add_argument('--llvm-enable-projects', metavar='OPT', action='append', default=None,
+                        help='enable llvm projects to build (in quotation marks separated by semicolons e.g.: "clang;openmp")')
+  buildopt.add_argument('--use-ccache', action="store_true", default=False,
+                        help='Using ccache during the build (default: %(default)s)')
+  arguments = parser.parse_args()
+  return arguments
+
+def generate_buildoptions(arguments):
+  install_root = Path(arguments.install_prefix)
+
+  base_cmake_args = [
+    f'-DCMAKE_INSTALL_PREFIX={install_root.as_posix()}',
+    f'-DCMAKE_BUILD_TYPE={arguments.build_type}',
+    f'-DCMAKE_TOOLCHAIN_FILE={arguments.toolchain}'
+  ]
+  generator = 'Ninja' if sys.platform == 'win32' else 'Unix Makefiles'
+  base_cmake_args.append(f'-G{generator}')
+
+  if arguments.use_ccache:
+    base_cmake_args.append('-DCMAKE_C_COMPILER_LAUNCHER=ccache')
+    base_cmake_args.append('-DCMAKE_CXX_COMPILER_LAUNCHER=ccache')
+
+  if arguments.cmake_param:
+    base_cmake_args.extend(arguments.cmake_param)
+
+  return base_cmake_args
+
+def normalize_builddir(project_srcdir, builddir, clean):
+  build_path = ''
+  if os.path.isabs(builddir):
+    build_path = os.path.join(builddir, os.path.basename(project_srcdir))
+  else:
+    build_path = os.path.join(project_srcdir, builddir)
+
+  if clean and os.path.exists(build_path):
+    shutil.rmtree(build_path)
+
+  return build_path
+
+def configure_llvm(arguments):
+  build_path = normalize_builddir(
+    settings.LLVM_DIR, arguments.builddir, arguments.clean)
+
+  build_options = generate_buildoptions(arguments)
+  cmake_cmd = ['cmake', '-B', build_path, '-S', settings.LLVM_DIR]
+  additional_options = [
+    f'-DLLVM_TARGETS_TO_BUILD={arguments.target}',
+    '-DLLVM_ENABLE_CLASSIC_FLANG=ON',
+  ]
+  if(arguments.llvm_enable_projects):
+    additional_options.append('-DLLVM_ENABLE_PROJECTS='+'; '.join(['%s']*len(arguments.llvm_enable_projects)) % tuple(arguments.llvm_enable_projects))
+    
+  build_options.extend(additional_options)
+  cmake_cmd = ['cmake', '-B' + build_path, '-H' + os.path.join(settings.LLVM_DIR, 'llvm')]
+
+  cmake_cmd.extend(build_options)
+
+  subprocess.run(cmake_cmd, check=True)
+

```

Comment:
Are the blank lines here (lines 98, 100, 102) necessary? Too much wasted vertical space IMHO.